### PR TITLE
Feature 2295 review assignment percentages

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/reviews/periods/DatePickerField.css [hunk 1]
```diff
@@ -17,16 +17,3 @@
   align-items: center;
   margin-bottom: 1rem;
 }
-
-.reviewPeriodSection {
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/DatePickerField.css:21]: This CSS class is no longer used.

File: web-ui/src/components/reviews/periods/ReviewPeriodCard.css [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.review-period-card {
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/ReviewPeriodCard.css:1]: The UI that renders each review period in `ReviewPeriods.jsx` was factored out to create the new component `ReviewPeriodCard`. This is the CSS for the new component.

File: web-ui/src/components/reviews/periods/ReviewPeriodCard.jsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
+import PropTypes from 'prop-types';
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/ReviewPeriodCard.jsx:1]: This is the new component that was factored out of `ReviewPeriod.jsx`.

File: web-ui/src/components/reviews/periods/ReviewPeriodCard.jsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  const [expanded, setExpanded] = useState(false);
+
+  const csrf = selectCsrfToken(state);
+  const currentMembers = selectCurrentMembers(state);
+  const canUpdateReviewAssignments =
+    selectHasUpdateReviewAssignmentsPermission(state);
+  const periods = selectReviewPeriods(state);
+
+  const period = periods.find(period => period.id === periodId);
+  const showPercentages =
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/ReviewPeriodCard.jsx:72]: This determines whether we should show percentages. The user must have a certain permission AND the review period must have a status of "AWAITING APPROVAL".

File: web-ui/src/components/reviews/periods/ReviewPeriods.jsx [hunk 5]
```diff
@@ -1,14 +1,6 @@
 import PropTypes from 'prop-types';
 import React, { useCallback, useContext, useEffect, useState } from 'react';
 
-import {
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/ReviewPeriods.jsx:4]: All the deleted code was moved to the new component `ReviewPeriodCard`.

File: web-ui/src/components/reviews/periods/ReviewPeriods.jsx [hunk 6]
```diff
@@ -135,7 +119,7 @@ const ReviewPeriods = ({ onPeriodSelected, mode }) => {
     closeDate: null
   });
   const [reviewStatus, setReviewStatus] = useState(ReviewStatus.CLOSED);
-  const [selfReviews, setSelfReviews] = useState(null);
+  const [selfReviews, setSelfReviews] = useState({});
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/periods/ReviewPeriods.jsx:122]: This holds a map of review data. Initializing it to an empty object simplifies the code a bit.

**Review by vhscom — APPROVED** [review]

**Review by jackkeller — APPROVED** [review]
:shipit:

**Pull request merged**